### ISSUE TITLE
CtrlSF's write buffer is corrupted

### ISSUE THREAD

**Issue by bfwg** [dyng/ctrlsf.vim#65]
When I try to edit the file which open by CtrlSF, I got a message says it's corrupted and I can not modify the file. What is happening to my CtrlSF

**Comment by dyng** [comment]
In short, nothing really really bad is happening, your change have just made CtrlSF's buffer can't be saved properly, be aware of that you can't insert new lines or delete entire a file in CtrlSF. (if not, there may be a bug)

If you need more help, please note more details about your system rather than a plain complaint, such as:

1. os
2. vim
3. backend (ag/ack)
4. file and file structure
5. detailed steps to reproduce

**Comment by bfwg** [comment]
@dyng 
Thanks for the response, I did try to modify the file after using CtrlSF and I did not know I can't do that.
So the correct way to use CtrlSF search is after find the result in a file using CtrlSF, go to the real file location and then modify it?

**Comment by dyng** [comment]
No, you can edit in CtrlSF window and save those changes to files (as shown in demo), if things don't go well, it may be a bug.

If you found details are difficult to describe, you can take some snapshots, like OP in this issue: #55

**Comment by bfwg** [comment]
@dyng 
Everything worked fine now, but I can still randomly get the buffer is corrupted error when I save in CtrlSF. 
I saw in the demo and you were using Macvim. I currently have a problem with Macvim. I have the nice plugin YouCompleteMe(YCM), every time I use `:q` to quit macvim and it will not trigger VimLeave so the YCM process will not be closed. If I keep closing with ':q or :wq' the processes will pile up.
![image](https://cloud.githubusercontent.com/assets/12819525/9402997/a9bd0aa0-4797-11e5-9635-f3ee977f810b.png)
So for now I'm closing Macvim using Command + q. This however will issue a VimLeave.
Is there a way I can use good old `:q` to close macvim?
I'll try to reproduce the corrupted buffer problem and I'll let you know if I've got it.

Thank you so much!
